app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -250,6 +256,10 @@ class SearchFragment : Fragment(), BackHandler {
         when (requestCode) {
             REQUEST_CODE_CAMERA_PERMISSIONS -> qrFeature.withFeature {
                 it.onPermissionsResult(permissions, grantResults)
+
+                if (context!!.isPermissionGranted(Manifest.permission.CAMERA)) {
```

Comment:
Could we use the safe call operator instead? It's not worth crashing over this if the result arrives as we're leaving the fragment.

Reply:
Just merged and then saw this - opened https://github.com/mozilla-mobile/fenix/pull/2104 as a follow up